### ISSUE TITLE
Add API to update ZK metadata of a segment

### ISSUE THREAD

**Issue by harold-kfuse** [apache/pinot#10458]
Looking at `BaseSingleSegmentConversionExecutor`, I see that it always uploads the segment, even if the segment did not change but only the metadata (e.g., updating custom map for some minion task ran on the segment). The controller then checks that the segment did not change and only updates ZK.

The network transfer of the segment from minion task to controller seems wasteful in this case, since the minion task should already know that the segment did not change. If there is an API to simply update ZK metadata, then that will be more efficient.

**Comment by Vaas-Arora** [comment]
Hi, can I pick this up?

**Comment by maahir22** [comment]
@harold-kfuse Can you point me to the controller where the unchanged segment check is made? I am unable to find it.

**Comment by heatclub** [comment]
@harold-kfuse, based on my analysis I see that 

1. [PinotSegmentUploadDownloadRestletResource](https://github.com/apache/pinot/blob/d8315c6025fbe963a9c01c08e0b021e54ae318d9/pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java#L280) already has code to handle segment METADATA, I assume the same endpoint can be used for `BaseSingleSegmentConversionExecutor` also
2. [BaseMultipleSegmentsConversionExecutor](https://github.com/apache/pinot/blob/d8315c6025fbe963a9c01c08e0b021e54ae318d9/pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java#L355) handles segment METADATA upload, the same can be extended in `BaseSingleSegmentConversionExecutor` for METADATA upload 

Please feel free to correct me if wrong.

**Comment by Jackie-Jiang** [comment]
@heatclub Ideally we don't need to upload anything when the segment remains exactly the same. I guess the proposal is to add an API to only modify some custom fields in the segment ZK metadata

**Comment by heatclub** [comment]
Thanks @Jackie-Jiang , I was able to go through the code and get that part.
Let me share the PR once its ready.

**Comment by ceekay47** [comment]
Hi @Jackie-Jiang, I see that there hasn't been any update on this ticket for a while. Can I look into this if @heatclub isn't working on it?

**Comment by heatclub** [comment]
Hey @ceekay47 please feel free to pick this up

**Comment by deadb0d4** [comment]
Hey, @ceekay47 can I look into this if you aren't working on it? Also, @Jackie-Jiang is this issue still needed?

**Comment by Jackie-Jiang** [comment]
Yes, the issue still applies. We do have a workaround by calling the ZK write rest API, but having a dedicate API and perform some validation is preferred

**Comment by goutamadwant** [comment]
@Jackie-Jiang I have an enhancement with #19180.

It adds a validated metadata-only endpoint using the expected segment CRC and a versioned ZK write. When a single-segment task returns the original segment unchanged, the minion now updates only the custom metadata map instead of creating and uploading a tarball. The existing upload path remains as a fallback for older controllers. Let me know if any suggestions. thanks!
